tests/cli/deploy.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { getMempoolTxns } from '../utils/network-utils.mjs';
+import { checkZkAppDeploymentResults } from '../utils/validation-utils.mjs';
+
+test.describe('zkApp-CLI', () => {
+  // TODO: https://github.com/o1-labs/zkapp-cli/issues/454
+  test(`should not deploy zkApp if not within the project dir, @parallel @smoke @deployment @fail-cases`, async () => {
+    const cliArg = 'deploy local';
+    const { execute, cleanup, path } = await prepareEnvironment();
+    console.info(`[Test Execution] Path: ${path}`);
+
+    try {
+      const { code, stdout } = await execute('zk', cliArg);
+      console.info(`[CLI StdOut] zk ${cliArg}: ${JSON.stringify(stdout)}`);
+
+      expect(code).toBe(0);
+      expect(stdout.at(-1)).toContain(
+        "config.json not found. Make sure you're in a zkApp project directory."
+      );
+    } finally {
+      await cleanup();
+    }
+  });
+
+  test(`should not deploy zkApp if no deployment alias found, @parallel @smoke @deployment @fail-cases`, async () => {
+    const projectName = crypto.randomUUID();
+    const { spawn, execute, cleanup, path } = await prepareEnvironment();
+    console.info(`[Test Execution] Path: ${path}`);
+
+    try {
+      await test.step('Project generation', async () => {
+        await generateProject(projectName, 'none', true, spawn);
+      });
+      await test.step('ZkApp deployment failure attempt (no aliases provided)', async () => {
+        const cliArg = 'deploy';
+        const { code, stdout } = await execute(
+          'zk',
+          cliArg,
+          `./${projectName}`
+        );
+        console.info(`[CLI StdOut] zk ${cliArg}: ${JSON.stringify(stdout)}`);
+
+        expect(code).toBe(0);
+        expect(stdout).toContain('No deploy aliases found in config.json.');
+      });
+      await test.step('ZkApp deployment failure attempt (unknown alias provided)', async () => {
+        const cliArg = `deploy ${crypto.randomUUID()}`;
+        const { code, stdout } = await execute(
+          'zk',
+          cliArg,
+          `./${projectName}`
+        );
+        console.info(`[CLI StdOut] zk ${cliArg}: ${JSON.stringify(stdout)}`);
+
+        expect(code).toBe(0);
+        expect(stdout).toContain('Deploy alias name not found in config.json.');
+      });
+    } finally {
+      await cleanup();
+    }
+  });
+
+  test(`should not deploy zkApp if aborted, @serial @smoke @deployment`, async () => {
+    const { spawn, cleanup, path } = await prepareEnvironment();
+    console.info(`[Test Execution] Path: ${path}`);
+
+    try {
+      await test.step('ZkApp project generation, configuration and deployment cancellation', async () => {
+        const mempoolTxns = await getMempoolTxns();
+        const { exitCode, stdOut } = await deployZkApp(
+          path,
+          'none',
+          true,
+          spawn,
+          true
+        );
+
+        expect(exitCode).toBeGreaterThan(0);
+        expect(stdOut).toContain('Aborted. Transaction not sent.');
```

Comment:
Regarding this and similar concerns. We should not treat this tests as something separate or too complicated to deal with. These are an integral part of development process, exercising the way how end users will see the application and interact with it. Hence, if you change something in the main codebase then you should run e2e tests alongside with the unit tests locally. You don't need to do any extra steps to run these tests locally. You of course can provide live env. but if it won't be found then mocked env. will be used. Then, all of the tests are isolated, they don't depend on each other and their titles are annotated with `@<tag>`, hence, you can simply [--grep](https://playwright.dev/docs/test-cli) needed tests to run only some portion of complete testing suite. If tests will fail after changes made then you will be able to see what exactly gone wrong (in terminal or in generated report).
So in the end I don't think relying on something that can possible be changed (like text) is not good approach because we want to also make sure that users will actually see what you supposed to output via the code execution.